packages/slate/src/utils/path-utils.js

```diff
@@ -1,14 +1,43 @@
 import { List } from 'immutable'
 
 /**
- * Compare paths `path` and `b` to see which is before or after.
+ * Compare paths `path` and `target` to see which is before or after.
  *
  * @param {List} path
- * @param {List} b
- * @return {Number|Null}
+ * @param {List} target
+ * @return {Number}
  */
 
 function compare(path, target) {
+  const m = min(path, target)
+
+  for (let i = 0; i < m; i++) {
+    const pv = path.get(i)
+    const tv = target.get(i)
+
+    // If the path's value is ever less than the target's, it's before.
+    if (pv < tv) return -1
+
+    // If the target's value is ever less than the path's, it's after.
+    if (pv > tv) return 1
+  }
+
+  // Otherwise size decides.
```

Comment:
Why does the length decide here? Is there a specific reason that we need that to end up that way?

What does it mean for `[0]` to be "before" `[0, 1]`? I guess it kind of makes sense in the top-down tree thinking. But I'm wondering if it would be better to either return `null` or `0`? Just curious.

Reply:
Good call, I actually don't know why I decided that a node should "come before" a descendant. Personally I would probably prefer to return `null` so that it's more apparent that it is not a normal scenario and when I use the method I can handle it separately if needed. What do you think?